packages/beeq/src/components/tag/bq-tag.tsx

```diff
@@ -0,0 +1,233 @@
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private handleHide = () => {
+    const ev = this.bqHide.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = false;
+    }
+  };
+
+  private handleShow = () => {
+    const ev = this.bqShow.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = true;
+    }
+  };
+
+  private handleClick = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    const ev = this.bqClick.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.selected = !this.selected;
+    }
+  };
+
+  private handleFocus = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    this.bqFocus.emit(this.el);
+  };
+
+  private get iconSize(): number {
+    return SIZE_TO_VALUE_MAP[this.size] || SIZE_TO_VALUE_MAP.medium;
+  }
+
+  private handleOnKeyDown = (event: KeyboardEvent) => {
+    this.bqKeyDown.emit(event);
+  };
+
+  private handleSlotChange = () => {
+    this.hasPrefix = hasSlotContent(this.prefixElem, 'prefix');
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <Host
+        class={{ 'is-hidden': !this.open }}
+        aria-hidden={!this.open ? 'true' : 'false'}
+        hidden={!this.open ? 'true' : 'false'}
+      >
+        <div
+          class={{
+            'bq-tag': true,
+            disabled: this.disabled,
+            active: !this.disabled && this.selected,
+            [`bq-tag__wrapper--${this.size} font-medium leading-regular`]: true,
+            [`bq-tag__${this.type}__${this.variant}`]: this.hasColor,
+          }}
+          onClick={this.handleClick}
+          onFocus={this.handleFocus}
+          onKeyDown={this.handleOnKeyDown}
+          role="button"
+          part="wrapper"
+          tabindex={this.disabled ? '-1' : '0'}
+        >
+          <span
+            class={{ 'inline-flex': true, '!hidden': !this.hasPrefix }}
+            ref={(spanElem) => (this.prefixElem = spanElem)}
+            part="prefix"
+          >
+            <slot name="prefix" onSlotchange={this.handleSlotChange} />
+          </span>
+          <div
+            class={{
+              'text-xs': this.size === 'extra_small',
+              'text-s': this.size === 'small',
+              'text-m': this.size === 'medium',
+            }}
+            part="text"
+          >
+            <slot />
+          </div>
+          {this.isRemovable && !this.hasColor && !this.disabled && (
+            <bq-button
+              class="bq-tag__close"
+              appearance="text"
+              size="small"
+              onClick={() => this.hide()}
+              part="btn-close"
+            >
```

Comment:
We can use directly `this.handleHide` in the onClick event handler

```suggestion
            <bq-button class="bq-tag__close" appearance="text" size="small" onClick={this.handleHide} part="btn-close">
```